### ISSUE TITLE
IFU changes to Extract1D

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5361]
_Issue [JP-1730](https://jira.stsci.edu/browse/JP-1730) was created on JIRA by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw):_

As detailed in multiple recent tickets linked below, there are problems with the Extract1D code as it relates to both MIRI MRS and NIRSpec IFU modes.

These issues were discussed and a resolution reached at a meeting 9/24/20, the outcome of which is detailed below.  (Attendees: [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) , [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) , [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) , [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe) , [James Muzerolle](https://jira.stsci.edu/secure/ViewProfile.jspa?name=muzerol) , [Anton Koekemoer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=koekemoe) , [Misty Cracraft](https://jira.stsci.edu/secure/ViewProfile.jspa?name=cracraft) , [Beth Sargent](https://jira.stsci.edu/secure/ViewProfile.jspa?name=sargent) , [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) )
----
**Background**:

At present, the Extract1d reference files for both MIRI MRS and NIRSpec IFU do not contain any information about the aperture radius to use for 1d spectral extraction.  As a result, the Extract1d code defaults to use 1/4 of the FOV for POINT source observations (https://jira.stsci.edu/browse/JP-1615), which is neither desirable nor in keeping with <https://outerspace.stsci.edu/pages/viewpage.action?spaceKey=JWSTCC&title=Vanilla+Spectral+Extraction].>  However, the JSON files used as Extract1D reference files are not able to (easily) contain the vector of wavelength dependent information necessary to describe the intended extraction radius.  This broadly varies between grating settings and/or bands, but since the MRS in particular is capable of making cubes from the full 5-28 micron interval a wavelength dependence to the radius is required.

The previous solution discussed via comments on <https://jira.stsci.edu/browse/JP-1049> for the MRS was to add the extra vectors into the aperture correction reference file.  This solution was not ideal though as it would require a different file to set the extraction details for IFU vs other modes.  As the code does not yet support any approach, this is our opportunity to work toward the solution that best implements the agreed-upon algorithm while simplifying the pipeline structure/documentation as much as possible.
----
**Solution**:

1) We will create a new FITS-format reference file for the Extract1D step.  Following the convention established for the CUBEPAR files, the 1st extension of this file will contain the parameter-reference-file type information currently in the JSON files (e.g., whether to subtract the background, what extraction method to use, etc).  The second extension will contain a FITS binary table of vectors specifying minimally the WAVELENGTH and RADIUS at each wavelength of the extraction aperture, along with the INNER_BKG, and OUTER_BKG radii of the background annulus.  Per <https://jira.stsci.edu/browse/JP-1563,> this background computation populates data used for the later Master Background step in Spec3, and should therefore be specified in the reference file.  [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) will create a draft reference file and iterate with [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) as necessary during development.  [James Muzerolle](https://jira.stsci.edu/secure/ViewProfile.jspa?name=muzerol) will create a corresponding version for NIRSpec prior to delivery of the reference files to CRDS.  Note that the NIRSpec file may include multiple extensions for the different grating settings.

2) We will modify the MIRI MRS FITS Aperture Correction reference file to remove the unnecessary columns that (1) migrates to the new Extract1D reference file.  At the same time, these files will be modified to add an extra radial dimension so that the interpolation of aperture correction factors can be performed over both wavelength and aperture radius (to support users forcing the pipeline to run with a custom extraction radius).  Placeholder values will be used for the radial interpolation at the present time, and can be more fully populated in flight, with best ground-test values in place for the default extraction radius.  This will bring the format of the MIRI file in line with the NIRSpec reference file, requiring an update to the data model and to <https://jira.stsci.edu/browse/JP-1049> as necessary.  [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) will contruct a draft reference file to iterate on with [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) .

3) We will modify the Extract1D code for both MRS and NIRSpec IFU modes to read in the radially-dependent vectors from the new Extract1D reference file, and use these radially dependent values within the wavelength loop over the input data cubes.  This will address <https://jira.stsci.edu/browse/JP-1615>

4) We will modify the IFU aperture correction code in Extract1D so that it correctly interpolates over the wavelength and radial dimensions (current only the radial interpolation is implemented).  This will address <https://jira.stsci.edu/browse/JP-1633>

5) The Extract1D code should be modified to always compute the background values in IFU mode, regardless of whether or not that background is to be subtracted internal to Extract1D.  This will require some changes to the Extract1D output data model as well, as detailed in the Cal WG conclusion to the discussion on <https://jira.stsci.edu/browse/JP-1563?focusedCommentId=476585&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-476585>  This is to support the fact that later fixes to Spec3 Master Background bugs for IFU mode require tweaks to the Extract1D results provided as input.
----
**Related Information**:

- As detailed in <https://jira.stsci.edu/browse/JP-1615,> values for wavelength, size/radius, and apcorr from the aperture correction reference file are read in by apply_apcorr.py. Since the input size/radius is in units of arcsec, apply_apcorr.convert_size_units converts it to units of spaxels using assign_wcs.util.compute_scale. However, compute_scale returns the scale in degrees where convert_size_units assumes that the value was returned in arcsec, therefore the radius is given as about 13,000 pixels instead of about 3 pixels.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=486616#comment-486616):_

Consultation with CRDS experts indicates that it should be possible to have a mix of EXTRACT1D ref files in json and FITS format within CRDS (with the possibility of a few little bumps associated with automatic file naming upon ingest). At the same time, the initial reaction from "that side of the house" was a universal question of why not store them as ASDF instead of FITS (owing to the complications of dealing with arrays within FITS BINTABLE and the interface to jwst datamodels). The existing (simple) EXTRACT1D ref files could easily be converted to an equivalent simple ASDF format, which would then allow all EXTRACT1D ref files to be the same format. For now, the appropriate path is to perhaps let [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) go ahead and create the MRS FITS version, at which point SCSB developers can review it to see how those same contents might fit within an ASDF form of the ref file.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=486711#comment-486711):_

Ah, that discussion sounds familiar.  As I recall last time it came up I wasn't convinced that ASDF was simpler for end-users to have to deal with than regular FITS, but I agree that we can punt that discussion for now.

[Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison): I've created an initial draft of an extract1d reference file, and this actually gives us a good opportunity to test out the workflow that we've been developing for delivery of new reference files during commissioning.  The upshot is that we have a shared Box folder that both the local MIRI team and the MIRI IDT can access at <https://stsci.app.box.com/folder/121562723577>

There are a variety of folders within this; you should have full access to all of them, and you can find the draft file at <https://stsci.app.box.com/folder/123396913795>

When it comes time to deliver the files to CRDS we'll move them into an area of that Box tree that we set up so that folks outside the MIRI team can see it.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=487352#comment-487352):_

[Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) That Box folder should now contain draft versions of the new extract1d and revised aperture correction reference files.  The latter has been updated to remove the background information columns, and more generally homogenize the format with the other instruments.  I.e., it has both a radius and a wavelength dimension to interpolate over, even if the radius interpolation is largely a placeholder for possible future functionality (so the aperture correction factors are the same at a given wavelength right now regardless of the radius called).  I've also added columns for channel and band set to ALL in keeping with the other instrument files; in future this allows us to potentially grow by having band-specific correction factors if that turns out to be sensible.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=489377#comment-489377):_

[David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) I am trying to figure out how to do the apcor with the new file. The apcor depends on the wavelength and radius. So what do I use to interpolate in radius ? The ifu.extract_ifu returns npixels = aperture_area, but we want the aperture radius not area so is this not the extract1d radius at the wavelength we are at. I don't think I have to calculate the radius but pass on the radius information from the ifu_extract_ifu step. Does this sound correct ?

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=491543#comment-491543):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) It looks like the work for this ticket is starting to converge, so I wanted to ask how we should handle the potential shift to calpars files described by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) .  As a placeholder to allow us to move forward, when I created the new MRS Extract1D reference file above I made one of the extensions contain the basic information about how to run the step (i.e., subtract_background = True).  This is exactly the kind of information that would be ideal for Jonathan's new calpars files, but how should we handle that change?

Should we keep everything in the FITS files that I created for now, and handle moving anything relevant to a calpars file at some future date with a dedicated ticket?  Or should we integrate that work into the current effort as a pathfinder for the calpars effort since we're in the middle of reworking the Extract1D reference file formatting already?  Let me know how you'd like to proceed.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=491545#comment-491545):_

I typically like to take the conservative approach of doing one step at a time, testing it, then moving on. So I'd say to just keep using the param settings from the new extract1d ref file for this work and then we'll do another small project to put them into a calpars ref file, once [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) has finished the infrastructure rework he's currently doing on that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=491546#comment-491546):_

I typically like to take the conservative approach of doing one step at a time, testing it, then moving on. So I'd say to just keep using the param settings from the new extract1d ref file for this work and then we'll do another small project to put them into a calpars ref file, once [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) has finished the infrastructure rework he's currently doing on that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=491856#comment-491856):_

Agree with the approach above. The other option is to make an explicit cfg file for this mode. The calpars work will be done this sprint, though further buy-in from ins/redcat should happen before full deployment. But no longer than another sprint..

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=495447#comment-495447):_

The details on how to make the new extract1d and apcorr reference files can be found in JP-1749.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=496289#comment-496289):_

The remaining issue is to get new reference files into CRDS for MIRI and NIRSpec for the extract1d and apcorr.
[David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) I believe the MIRI reference files are ready - is that correct ?

[James Muzerolle](https://jira.stsci.edu/secure/ViewProfile.jspa?name=muzerol) I realize proposal time is taking up most peoples time. In ████████████████████████████████ is the code the nirspec team can start with. If you can point me to the person who will make these files I can explain what I have done so far. 

convert_ifu_extract1d_datamodels.py takes jwst_nirspec_extr1d_0001.fits and converts it to  jwst_nirspec_extr1d_0001_dict.asdf. 

convert_ifu_nirspec_apcorr_dict.py takes jwst_nirspec_apcorr_0004.fits and pulls out the IFU apcorr info to make jwst_nirspec_apcorr_0004_dict.asdf. The missing piece is making the table for the generic case - the first table.

**Comment by stscijgbot-jp** [comment]
_Comment by [James Muzerolle](https://jira.stsci.edu/secure/ViewProfile.jspa?name=muzerol) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=496291#comment-496291):_

[Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) Thanks very much for providing the code! [Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) will be the one handling the file creation & submission.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=496292#comment-496292):_

Yes, final MIRI files are ready to deliver.  I've staged them in the Delivery section of the MIRI Box folder that we'll be using for deliveries during commissioning at <https://stsci.app.box.com/folder/126338587373>

Let me know when I should trigger a CRDS ticket.

I'll add people to the viewer list for this Box folder as necessary (added [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) and [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) already).

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=498063#comment-498063):_

Bumping this ticket- is everything in place to move ahead with delivering the new reference files and merging #5383 ?

**Comment by jemorrison** [comment]
Just waiting on NIRSPEC reference files. Graham Kanarek is working on them. @gkanarek  are you also making the extract1d NIRSPEC ref file or just using the one I made ?

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=499059#comment-499059):_

Updating this ticket that the new pull request is at #5506 (replacing the previous PR 5383).

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=499065#comment-499065):_

Filed corresponding CRDS ticket at <https://jira.stsci.edu/browse/CRDS-394>

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=499232#comment-499232):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) as you can see from the above comment - David has filed a CRDS ticket to get the MIRI Apcor and Extract1D files in CRDS. Can you file a CRDS for the NIRSPEC files. Since there are so many NIRSPEC apcorr files you might want to file two CRDS tickets. Once we get those into CRDS then we can merge this PR and get the code it into Build 7.7.

**Comment by stscijgbot-jp** [comment]
_Comment by [Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=499311#comment-499311):_

Hi [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) I can submit the files, sure. I'll make one change to your files before submitting -- I'm going to remove NRS_MSASPEC from exp_type in the metadata, because these are going to be IFU files only, and we'll use the original file for the MSA. I'll talk to Matt McMaster about how to make sure the rmap selects the correct file.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500028#comment-500028):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) just checking on the submitting the new reference files. I want to make sure we sync the reference files and this PR together. MIRI has CDRS-394 for these reference files if you need to see how they are doing it.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500633#comment-500633):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) David Law has submitted the new MIRI extract1d and apcorr reference files. He learned a few things about the format of the EXP_TYPE and DATAMODL [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) could you fill in what Graham needs to do. We want to get these files in to CRDS - it possible - Wednesday or Thursday so we can finalize any last minute changes and get the code merged on Friday.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500648#comment-500648):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) You can see the changes that I had to make at <https://github.com/STScI-MIRI/miri3d/commit/0eb1c26c4968f4ae54de682a6f8305df44196779#diff-6dc317b1f7990c8484e08dbe276a964956fbfa5059f5d3893928ed836ddca261>

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500787#comment-500787):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) are you around ? We need to get the NIRSPEC apcor and extract1d reference files delivered to CRDS today if possible. If not let me know we have to figure out if this improvement is part of Build 7.7

**Comment by stscijgbot-jp** [comment]
_Comment by [Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500800#comment-500800):_

Hi Jane, I'm waiting on CRDS to be updated so I can deliver the files, as per CRDS-395. I'm not sure what the timeline is for that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=500832#comment-500832):_

FYI: Watch CRDS-395 for updates. Also, if the new references can be attached or linked to from that issue, that will help in testing.

**Comment by hbushouse** [comment]
Fixed by #5506

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502189#comment-502189):_

[Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) Any update on the NIRSpec delivery?

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502203#comment-502203):_

[Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham)  [Robert Jedrzejewski](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rij)  [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) [Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) I probably need to open a new ticket - but for now I will outline what is needed.

We need a new rmap for the NIRSPEC extract1d. For IFU data the json file needs to be replaced with the correct asdf file.  I just need pointers on what I is needed. I can download the current rmap and update the filename to the asdf file. I do not know how this process works. Where do I up load the new updated rmap and do I just  change the rmap from jwst_nirspec_extract1d_0010.rmap to jwst_nirspec_extract1d_0011.rmap ?

For the new Apcorr rmap for NIRSPEC we need to add a selection for filter and grating for IFU data. I see how to do this by looking at other rmaps that have a filter and grating selection. I can update the rmap but again I am not sure where to upload the new files. 
 Do I need an account "dstring"  to test this ?

**Comment by stscijgbot-jp** [comment]
_Comment by [Robert Jedrzejewski](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rij) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502208#comment-502208):_

Looks like the extract1d reference file selects on exp_type only, so delivering new reference files will take care of the rmap creation.

For the apcorr reference file, the selection is now on exp_type, filter and grating (used to be just exp_type).  I'm not certain about this, but I think that if the CRDS spec has been updated to include the new PAR_KEYs, then submitting new reference files with the EXP_TYPE, FILTER and GRATING keywords should also take care of creating the corresponding rmap?  You could try this by submitting 1 new apcorr reference file to CRDS TEST and seeing if the rmap is updated to what you would expect.  You should be able to submit a new apcorr reference file to TEST using the regular extended batch submit form, just select the TEST server: <https://jwst-crds-test.stsci.edu/>

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502217#comment-502217):_

CCD-891 is done and this ticket 'Updated NIRSpec APCORR spec to include FILTER and GRATING"
This was finished Dec 28. 



CRDS-395 (marked in process) includes a delivery of the new NIRSPEC files. But there is also a CRDS-402 opened on Dec 28 that I believe was closed by CCD-891. In the comments Jonathan states:
 Both TEST and OPS have CRDS 10.3.2 installed, enabling certification. rmap updates and ingest may now proceed.

So how is a rmap created ? That part I do not understand. The rmaps in <https://jwst-crds-test.stsci.edu](https://jwst-crds-test.stsci.edu/)> are not up-to-date.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502220#comment-502220):_

[Robert Jedrzejewski](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rij)  Re-reading the message I think if Graham re-submits his reference files (which is what CRDS-402 is about) then the rmap should be updated. [Graham Kanarek](https://jira.stsci.edu/secure/ViewProfile.jspa?name=gkanarek) can you do that.

Graham  attached the files in CRDS-395 so if he is not around I can submit them - I just never have submitted a file to CRDS so I don't know how to do that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=502878#comment-502878):_

Fixed by #5506

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=503083#comment-503083):_

David this is ready for testing

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-1730?focusedCommentId=505444#comment-505444):_

Appears to be working as intended, this ticket can be closed.  Shifting focus now to consider if <https://jira.stsci.edu/browse/JP-1563> can be updated now that these changes are in place.
